### DISCUSSION TITLE
Consider using CSSOM to manage styles

### DISCUSSION THREAD

**Discussion by davletovalmir** [capricorn86/happy-dom#1130] [Ideas]
Currently, styles are not on par with CSSOM standards. The test below would fail:

```js
const node = document.createElement('div');
node.style.backgroundColor = "#ffa015";
expect(node.style.backgroundColor).toEqual("rgb(255, 160, 21)"); // <-- will fail here
```

Though it sounds like `node.style.backgroundColor` should return `"#ffa015"`, according to the specs, it must convert hexadecimal values to `rgb` format.

### References:
- [CSSOM CSSStyleDeclaration Interface](https://www.w3.org/TR/cssom-1/#the-cssstyledeclaration-interface)
- [CSSOM Color Level 3](https://www.w3.org/TR/css-color-3/) (search for **4.2. Numerical color values**)

**Comment by capricorn86** [comment]
This should be fixed.

I have created the feature #1829 for it.
